File: packages/youtube-player/android/src/main/java/io/capawesome/capacitorjs/plugins/youtubeplayer/classes/events/PlayerErrorEvent.java
package io.capawesome.capacitorjs.plugins.youtubeplayer.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.youtubeplayer.interfaces.Result;

public class PlayerErrorEvent implements Result {

    @NonNull
    private final String code;

    @NonNull
    private final String id;

    public PlayerErrorEvent(@NonNull String code, @NonNull String id) {
        this.code = code;
        this.id = id;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("code", code);
        result.put("id", id);
        return result;
    }
}

File: packages/youtube-player/ios/Plugin/Classes/Events/PlayerErrorEvent.swift
import Capacitor
import Foundation

@objc public class PlayerErrorEvent: NSObject, Result {
    let code: String
    let id: String

    init(code: String, id: String) {
        self.code = code
        self.id = id
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["code"] = code
        result["id"] = id
        return result as AnyObject
    }
}
